ngraph/frontend/paddlepaddle/src/op/pool2d.cpp

```diff
@@ -25,47 +15,232 @@ namespace ngraph
                     auto data = node.get_ng_input("X");
+
                     auto pooling_type = node.get_attribute<std::string>("pooling_type");
                     auto global_pooling = node.get_attribute<bool>("global_pooling");
                     auto adaptive = node.get_attribute<bool>("adaptive");
                     auto kernel_shape = node.get_attribute<std::vector<int32_t>>("ksize");
-                    if (pooling_type == "max" && !global_pooling)
+
+                    auto rounding_type = node.get_attribute<bool>("ceil_mode")
+                                             ? ngraph::op::RoundingType::CEIL
+                                             : ngraph::op::RoundingType::FLOOR;
+
+                    if (pooling_type.empty())
                     {
-                        auto strides = node.get_attribute<std::vector<int32_t>>("strides");
-                        auto paddings = node.get_attribute<std::vector<int32_t>>("paddings");
-                        auto rounding_type = node.get_attribute<bool>("ceil_mode")
-                                                 ? ngraph::op::RoundingType::CEIL
-                                                 : ngraph::op::RoundingType::FLOOR;
-                        return node.default_single_output_mapping(
-                            {std::make_shared<ngraph::opset6::MaxPool>(
-                                data,
-                                ngraph::Strides(strides.begin(), strides.end()),
-                                ngraph::Shape(paddings.begin(), paddings.end()),
-                                ngraph::Shape(paddings.begin(), paddings.end()),
-                                ngraph::Shape(kernel_shape.begin(), kernel_shape.end()),
-                                rounding_type)},
-                            {"Out"});
+                        pooling_type = "max";
                     }
-                    else if (pooling_type == "avg" &&
-                             (global_pooling ||
-                              (adaptive && all_of(kernel_shape.begin(),
-                                                  kernel_shape.end(),
-                                                  [](int32_t s) { return s == 1; }))))
+
+                    PDPD_ASSERT((pooling_type == "max") || (pooling_type == "avg"),
+                                "pool2d: not supported pooling type !");
+                    PDPD_ASSERT(kernel_shape.size() == 1 || kernel_shape.size() == 2,
+                                "pool2d: ksize must be 1 or 2!");
+
+                    PartialShape input_shape = data.get_partial_shape();
+                    PDPD_ASSERT(input_shape.rank().is_static(), "pool2d: X rank must be static!");
+                    int32_t input_rank = input_shape.rank().get_length();
+                    PDPD_ASSERT(input_rank >= 2, "kernel rank must be greater than 2");
+                    uint64_t input_h = input_shape[input_rank - 2].get_length();
```

Comment:
This code requires input shape to be static and it will not allow reshaping of the model. Please use ShapeOf op if you need to know shape of the input.

Reply:
@mvafin Hi the input shape are needed for computing kernel size / stride for PaddlePaddle pool2d.

1.  Adaptive Pool which are needed to compute the kernel/stride for actual pool. That parts needs the input shape to be static.
2.  If the `kernel shape` is greater than `input + padding`, the kernel shape will be floored to `input + padding` [Reference ](https://github.com/nosovmik/openvino/pull/53#discussion_r635893228)

Reply:
@mvafin Thanks for you advice, I remove the static rank check in the front. And only does this when we really need access to the input shape